e2e-tests/src/test/era_payout.rs

```diff
@@ -29,41 +30,55 @@ fn payout_within_two_block_delta(expected_payout: Balance, payout: Balance) {
     );
 }
 
-fn wait_to_second_era<C: ReadStorage>(connection: &C) -> EraIndex {
-    let active_era = get_active_era(connection);
+async fn wait_to_second_era(connection: &Connection) -> EraIndex {
+    let active_era = connection.get_active_era(None).await;
     if active_era < 2 {
-        wait_for_next_era(connection).expect("Era is active");
-        wait_for_next_era(connection).expect("Era is active");
+        connection.wait_for_n_eras(2, BlockStatus::Best).await;
     }
-    get_active_era(connection)
+    connection.get_active_era(None).await
 }
 
-fn force_era_payout(config: &Config) -> anyhow::Result<()> {
-    let root_connection = config.create_root_connection();
-    let active_era = wait_to_second_era(&root_connection);
-    wait_for_next_era(&root_connection)?;
+async fn force_era_payout(config: &Config) -> anyhow::Result<()> {
+    let root_connection = config.create_root_connection().await;
+    let active_era = wait_to_second_era(&root_connection.connection).await;
+    root_connection
+        .connection
+        .wait_for_n_eras(1, BlockStatus::Best)
+        .await;
     let active_era = active_era + 1;
 
     let starting_session = active_era * DEFAULT_SESSIONS_PER_ERA;
-    wait_for_session(&root_connection, starting_session + 1)?;
+    root_connection
+        .connection
+        .wait_for_session(starting_session + 1, BlockStatus::Best)
+        .await;
 
     // new era will start in the session `starting_session + 3`
-    staking_force_new_era(&root_connection, XtStatus::InBlock);
-    wait_for_session(&root_connection, starting_session + 3)?;
-
-    let payout = get_payout_for_era(&root_connection, active_era);
+    root_connection.force_new_era(TxStatus::InBlock).await?;
+    root_connection
+        .connection
+        .wait_for_session(starting_session + 3, BlockStatus::Best)
+        .await;
+
+    let payout = root_connection
+        .connection
+        .get_payout_for_era(active_era, None)
+        .await;
     let expected_payout = era_payout((3 * DEFAULT_SESSION_PERIOD) as u64 * MILLISECS_PER_BLOCK).0;
 
     payout_within_two_block_delta(expected_payout, payout);
 
     Ok(())
 }
 
-fn normal_era_payout(config: &Config) -> anyhow::Result<()> {
-    let connection = create_connection(&config.node);
+async fn normal_era_payout(config: &Config) -> anyhow::Result<()> {
+    let root_connection = config.create_root_connection().await;
 
-    let active_era = wait_to_second_era(&connection);
-    let payout = get_payout_for_era(&connection, active_era - 1);
+    let active_era = wait_to_second_era(&root_connection.connection).await;
+    let payout = root_connection
+        .connection
+        .get_payout_for_era(active_era - 1, None)
```

Comment:
What do you think about introducing some macro that would implement `StakingApi` for `RootConnection` and `SignedConnection`? (in general every API for `RootConnection` and `SignedConnection` and every `UserAPI` for `RootConnection`)

Reply:
ble